spt.c

```diff
@@ -78,15 +79,33 @@ notify_send(char *cmt)
 }
 
 void
-display_remaining_time(int remainingtime)
+display_state(int remainingtime, int suspend)
 {
-	char buf[17];
+	char *buf, *fmt;
+	int size = 0;
 
-	snprintf(buf, 17, "Remaining: %02d:%02d\n",
+	if (suspend) {
+		size = 29;
+		buf = malloc(29);
```

Comment:
@pips- Why do we need to use malloc for this string that is not very large, I seldom use heap memory but is there any benefit here?